mev_inspect/arbitrages.py

```diff
@@ -78,7 +83,49 @@ def _get_arbitrages_from_swaps(swaps: List[Swap]) -> List[Arbitrage]:
         ]
 
 
-def _get_all_start_end_swaps(swaps: List[Swap]) -> List[Tuple[Swap, Swap]]:
+def _get_shortest_route(
+    start_swap: Swap,
+    end_swaps: List[Swap],
+    all_swaps: List[Swap],
+) -> Optional[List[Swap]]:
+    for end_swap in end_swaps:
+        if start_swap.token_out_address == end_swap.token_in_address:
+            return [start_swap, end_swap]
+
+    other_swaps = [
+        swap for swap in all_swaps if (swap is not start_swap and swap not in end_swaps)
+    ]
+
+    if len(other_swaps) == 0:
+        return None
+
+    shortest_route_rest = None
```

Comment:
what is shortest_route_rest meant to indicate? It's a bit confusing, something along the lines current_shortest_route?

Reply:
it’s the shortest path through the rest of the nodes to one of the ends

Reply:
is it not used in this case as a the current of the current/next variable pair to find the shortest route recursively?

Reply:
i should add a type to it

it’s Optional[List[Swap]]

it holds the shortest path so far as we’re iterating through each of the potential next start nodes

high level is like

get shortest(start, ends)
- find all potential continuations
- for each continuation, get shortest path from that one to an end (get_shortest(next, ends))
- return start + shortest of all continuations

Reply:
shortest path so far -> current_shortest_path

This seems like a more descriptive name

Reply:
it's not current because it isn't the full path

maybe `shortest_next_path`

Reply:
wnet with "shortest_remaining_route"